hieroplatform/src/main/java/org/hiero/sketch/table/BaseRowSnapshot.java

```diff
@@ -0,0 +1,89 @@
+package org.hiero.sketch.table;
+
+import org.hiero.sketch.table.api.IRow;
+import org.hiero.utils.Converters;
+
+public abstract class BaseRowSnapshot implements IRow {
+    protected final Schema schema;
+
+    protected BaseRowSnapshot(Schema schema) {
+        this.schema = schema;
+    }
+
+    @Override
+    public int rowSize() {
+        return this.schema.getColumnCount();
+    }
+
+    @Override
+    public Schema getSchema() {
+        return this.schema;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if ((o == null) || !(o instanceof IRow)) return false;
+        IRow that = (IRow) o;
+        if(this.schema != that.getSchema())
+            return false;
+        for (String cn: this.schema.getColumnNames()) {
+            if (this.isMissing(cn) && that.isMissing(cn)) {
+                return true;
+            } else if (this.isMissing(cn) || that.isMissing(cn)) {
+                return false;
+            } else {
+                switch (this.schema.getKind(cn)) {
+                    case String:
```

Comment:
We have a new kind now, called Category. It should be used exactly as a String.
The only difference is that you can histogram categories, but not strings.